### ISSUE TITLE
General idea

### ISSUE THREAD

**Issue by alessandrokonrad** [Berry-Pool/cardano-web3#1]
## Intro

The idea of this library is to have a very high level JavaScript SDK to interact with Cardano and build transactions.
Right now the only way to interact with Cardano from the browser/Js is with the [serialization-lib](https://github.com/Emurgo/cardano-serialization-lib) (A library written in rust, but can be compiled to webAssembly and so be used in the browser). This library is very low level and close to the ledger specs, so it's not very developer friendly and requires some deeper understanding of Cardano.
However it also has advantages: It's really fast because of webAssembly and it's very flexible because it's so close to the ledger specs.
So it makes sense to use the serialization-lib as foundation and build on top of that one a high level API.

## Foundational work

The ground work I did so far is that the library can be seamlessly used as npm package for node.js and the browser. The library can also be built as a package directly for the web. WebAssembly always requires to be imported asynchronously, this is kinda painful, but with Webpack 5 you can enable `topLevelAwait` and import the whole module instead asynchronously. This makes everything much smoother again and the library can be imported as any other without having to care about async and webAssembly really.

## Goal

The goal/idea is to have a library that is very easy to use and understand. Work with bech32 encodings wherever it's possible (e.g. addresses), simple json objects instead of complicated data structures etc.
A good reference is the web3 library on [Ethereum](https://github.com/ChainSafe/web3.js). Instantiate a Web3 object and create transaction, interact with wallets, etc. from there.

**Comment by mauryaland** [comment]
You could be interested by the work started by the [RAY Network](https://raynetwork.io/) team few months ago:  https://github.com/ray-network/cardano-web3.js. It seems that the developments are paused right now but it may be worth taking a look!

**Comment by mauryaland** [comment]
I've got an answer from a RAY team member: "Yes the completed Web3.js library and documentation are scheduled for release this quarter". So their library should have some updates soon!

**Comment by alessandrokonrad** [comment]
Oh awesome! Okay then it's probably better to wait and see what they come up with.

**Comment by Riley-Kilgore** [comment]
I agree that it could make sense to wait, and another group that is doing similar work is MLabs in this repo: https://github.com/Plutonomicon/cardano-browser-tx 🤷‍♂️

**Comment by gerciljunio** [comment]
I'm too developing an package thats going work with multiple wallets and networks, with main focus in Cardano!
https://twitter.com/adanizecom/status/1497636212642926595
